torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp

```diff
@@ -211,8 +212,10 @@ void FixupONNXLoopNodeInputs(Node* node) {
 
   // add cast to condition input outside the loop.
   Value* cond_val = node->input(1);
-  if (IsCondCastRequired(cond_val))
-    InsertCastForCond(cond_val, graph, node);
+  if (IsCondCastRequired(cond_val)) {
+    auto* cast_node = InsertCastForCond(cond_val, graph, node);
```

Comment:
For these two I explicitly leave it outside, it may be different in terms of how the scope is set depending on where this function is being called: inserting before loop node, or inserting inside loop subblock.